packages/devtools_app/test/standalone_ui/ide_shared/property_editor_test.dart

```diff
@@ -2,80 +2,145 @@
+  late MockEditorClient mockEditorClient;
+  late PropertyEditorController controller;
+  late PropertyEditorView propertyEditor;
 
-      final stringInput = _findTextFormField('title');
-
-      expect(stringInput, findsOneWidget);
-    });
+  setUpAll(() {
+    setGlobal(ServiceConnectionManager, FakeServiceConnectionManager());
+    setGlobal(IdeTheme, IdeTheme());
+  });
 
-    testWidgets('double input', (tester) async {
-      await tester.pumpWidget(wrap(propertyEditor));
+  setUpAll(() {
+    mockEditorClient = MockEditorClient();
+    when(
+      mockEditorClient.activeLocationChangedStream,
+    ).thenAnswer((_) => eventStream);
 
-      final doubleInput = _findTextFormField('width');
+    controller = PropertyEditorController(mockEditorClient);
+    propertyEditor = PropertyEditorView(controller: controller);
+  });
 
-      expect(doubleInput, findsOneWidget);
-    });
+  group('on cursor location change', () {
+    void waitForEditableArgs(
+      List<EditableArgument> args, {
+      required Future then,
+    }) {
+      controller.editableArgs.addListener(() async {
+        final current =
+            controller.editableArgs.value.map((arg) => arg.name).toList();
+        final expected = args.map((arg) => arg.name).toList();
 
-    testWidgets('bool input', (tester) async {
-      await tester.pumpWidget(wrap(propertyEditor));
+        if (expected.every((arg) => current.contains(arg))) {
+          await then;
+        }
+      });
+    }
 
-      final boolInput = _findDropdownButtonFormField('softWrap');
+    setUp(() {
+      for (final MapEntry(key: location, value: result)
+          in locationToArgsResult.entries) {
+        when(
+          // ignore: discarded_futures, for mocking purposes.
+          mockEditorClient.getEditableArguments(
+            textDocument: location.$1,
+            position: location.$2,
+          ),
+        ).thenAnswer((realInvocation) => Future.value(result));
+      }
+    });
 
-      expect(boolInput, findsOneWidget);
-      await _verifyDropdownMenuItems(
-        boolInput,
-        menuOptions: ['true', 'false'],
-        selectedOption: 'true',
-        tester: tester,
+    testWidgets('properties match args for first cursor location', (
+      tester,
+    ) async {
+      // Wait for expected editable arguments and inputs:
+      waitForEditableArgs(
+        result1.args,
+        then: tester.runAsync(() async {
+          expect(_findNoPropertiesMessage, findsNothing);
+          expect(_findTextFormField('title'), findsOneWidget);
+          expect(_findTextFormField('width'), findsOneWidget);
+          expect(_findTextFormField('height'), findsOneWidget);
+        }),
       );
-    });
 
-    testWidgets('enum input', (tester) async {
+      // Load property editor then send an active location changed event:
       await tester.pumpWidget(wrap(propertyEditor));
+      eventController.add(activeLocationChangedEvent1);
```

Comment:
why send this at the end of the test?

Reply:
It's not **really* at the end of the test, this is what triggers the request for editable arguments, the existence of which we wait for in the `waitForEditableArgs` invocation above.